pkg/service/backup/restore_worker.go

```diff
@@ -152,11 +152,7 @@ func (w *restoreWorkerTools) ValidateTableExists(keyspace, table string) error {
 	defer q.Release()
 
 	var name string
-	if err := q.Scan(&name); err != nil {
-		return err
-	}
-
-	return nil
+	return q.Scan(&name)
```

Comment:
Well, the idea of having `Release` on gocql is to optimize memory allocations, so it would not need to create a new `gocql.Query` every time it needs it, but rather borrow it from the pool.
Having that said we need to call `Release` every time we get a query and done with using it.
So, answer is Yes, we need to call it there

Reply:
I have went over the code to double check if we doing that and found couple places where we miss the thing.
Please find them in seperate commit